model/flow/service_event.go

```diff
@@ -48,6 +49,16 @@ func (sel ServiceEventList) EqualTo(other ServiceEventList) (bool, error) {
 	return true, nil
 }
 
+func FilterServiceEvents[T *EpochSetup | *EpochCommit | *VersionBeacon](sel ServiceEventList) []T {
```

Comment:
Yes - but we keep the list of service events fixed and rarely changing. Additionally, and this is the main reason, this method already returns properly typed slice, otherwise caller would have to rewrite it to get a proper type